### PR TITLE
get pop to behave like pop

### PR DESCRIPTION
**Pull request by akaszynski**
This is a quick fix to a quirk I found recently where `pop` in `DataSetAttributes` wasn't behaving like the built-in dictionary pop.  See:

https://docs.python.org/3/library/stdtypes.html#dict.pop

We just need to add the `default` argument such that if the key isn't present in the `DataSetAttributes` instance, we return the default arg.

### CODE REVIEW

**Comment by JevinJ** [comment]
I've made some changes to avoid the separate sentinel, if you agree it's good.

**Review by JevinJ — APPROVED** [review]

File: pyvista/core/datasetattributes.py [hunk 1]
```diff
@@ -288,7 +291,12 @@ def pop(self, key):
             copy = vtk_arr.NewInstance()
             copy.DeepCopy(vtk_arr)
             vtk_arr = copy
-        self.remove(key)
+        try:
+            self.remove(key)
+        except KeyError:
+            if default in self.pop.__defaults__:
+                raise
```
Comment (akaszynski) [inline pyvista/core/datasetattributes.py:298]:
```suggestion
                raise KeyError
```

**Review by akaszynski — COMMENTED** [review]
Should raise a `KeyError` I think.  Otherwise, this works.

File: pyvista/core/datasetattributes.py [hunk 2]
```diff
@@ -288,7 +291,12 @@ def pop(self, key):
             copy = vtk_arr.NewInstance()
             copy.DeepCopy(vtk_arr)
             vtk_arr = copy
-        self.remove(key)
+        try:
+            self.remove(key)
+        except KeyError:
+            if default in self.pop.__defaults__:
+                raise KeyError
```
Comment (larsoner) [inline pyvista/core/datasetattributes.py:298]: I don't think you need the `KeyError`, the bare `raise` will raise the previous error which is already a `KeyError` (and might contain more useful information about what the problematic key was)

Comment (JevinJ) [reply pyvista/core/datasetattributes.py:298]: remove() can raise KeyError, through get_array(), and raise in an except re-raises the exception.

Comment (larsoner) [reply pyvista/core/datasetattributes.py:298]: I think we're saying the same thing. I'm suggesting that 70174ee should be reverted

Comment (akaszynski) [reply pyvista/core/datasetattributes.py:298]: Agreed.  Sorry, my bad suggestion!

Comment (akaszynski) [inline pyvista/core/datasetattributes.py:298]:
```suggestion
                raise
```
Lazily commiting here as I have other branches up...

**Review by banesullivan — APPROVED** [review]

**Pull request merged**